packages/tabsheet/src/vaadin-tabsheet-mixin.js

```diff
@@ -18,11 +18,33 @@ class TabsSlotController extends SlotController {
     super(host, 'tabs');
     this.__tabsItemsChangedListener = this.__tabsItemsChangedListener.bind(this);
     this.__tabsSelectedChangedListener = this.__tabsSelectedChangedListener.bind(this);
+    this.__tabIdObserver = new MutationObserver((mutations) => {
+      mutations.forEach((mutation) => {
+        const tab = mutation.target;
+
+        if (mutation.type === 'attributes') {
+          // We use attributeFilter to only observe ID mutation,
+          // no need to check for attribute name separately.
+          host.__linkTabAndPanel(tab);
+
+          if (tab.selected) {
+            host.__togglePanels(tab);
+          }
+        }
+      });
+    });
   }
 
   /** @private */
   __tabsItemsChangedListener() {
-    this.host._setItems(this.tabs.items);
+    const items = this.tabs.items || [];
```

Comment:
Should we call `this.__tabIdObserver.disconnect()` here to have any discarded tabs unobserved?

Reply:
Thanks, done.